src/pydocstyle/cli.py

```diff
@@ -0,0 +1,103 @@
+"""Command line interface for pydocstyle."""
+import logging
+import sys
+
+from .utils import log
+from .violations import Error
+from .config import ConfigurationParser, IllegalConfiguration
+from .checker import check
+
+
+__all__ = ('main', )
+
+
+class ReturnCode(object):
+    no_violations_found = 0
+    violations_found = 1
+    invalid_options = 2
+
+
+def run_pydocstyle(use_pep257=False):
+    log.setLevel(logging.DEBUG)
+    conf = ConfigurationParser()
+    setup_stream_handlers(conf.get_default_run_configuration())
+
+    try:
+        conf.parse()
+    except IllegalConfiguration:
+        return ReturnCode.invalid_options
+
+    run_conf = conf.get_user_run_configuration()
+
+    # Reset the logger according to the command line arguments
+    setup_stream_handlers(run_conf)
+
+    if use_pep257:
+        log.warning("Deprecation Warning:\n"
+                    "pep257 has been renamed to pydocstyle and the use of the "
+                    "pep257 executable is deprecated and will be removed in "
+                    "the next major version. Please use `pydocstyle` instead.")
+
+    log.debug("starting in debug mode.")
+
+    Error.explain = run_conf.explain
+    Error.source = run_conf.source
+
+    errors = []
+    try:
+        for filename, checked_codes in conf.get_files_to_check():
+            errors.extend(check((filename,), select=checked_codes))
+    except IllegalConfiguration:
+        # An illegal configuration file was found during file generation.
+        return ReturnCode.invalid_options
+
+    code = ReturnCode.no_violations_found
+    count = 0
+    for error in errors:
+        sys.stderr.write('%s\n' % error)
+        code = ReturnCode.violations_found
```

Comment:
Is it necessary to reassign this for every error? Why not do something like:

```py
count = 0
for error in errors:
    print('%s' % error)   # because we all agree sys.stderr isn't correct
    count += 1
if count > 0:
   code = ReturnCode.violations_found
else:
   code = ReturnCode.no_violations_found
```